src/content-script.ts

```diff
@@ -37,56 +37,60 @@ const injectScript = (path: string) => {
 
 export const injectSettings = (options: SimpleOptions) => {
   inject(script => {
-    const serialized = JSON.stringify(options);
+    if (document instanceof HTMLDocument) {
+      const serialized = JSON.stringify(options);
 
-    script.textContent = `this.treeRenderOptions = ${serialized};`;
+      script.textContent = `this.treeRenderOptions = ${serialized};`;
+    }
   });
 };
 
-browserSubscribeOnce(MessageType.FrameworkLoaded,
-  () => {
-    loadOptions().then(options => {
-      // We want to load the tree rendering options that the UI has saved
-      // because that allows us to send the correct tree immediately upon
-      // startup and send it to the message queue, allowing Augury to render
-      // instantly as soon as the application is loaded. Without this bit
-      // of code we would have to wait for the frontend to start and load its
-      // options and then request the tree, which would add a lot of latency
-      // to startup.
-      injectSettings(options);
-
-      injectScript('build/backend.js');
+if (document instanceof HTMLDocument) {
```

Comment:
This seems reasonable.

Reply:
Thank you @santiago-elustondo for reviewing my changes. I reverted the other change since it was indeed not needed.